frontend/app-development/features/appPublish/components/Deploy/DeployDropdown/DeployPopover/DeployPopover.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { type ReactElement, useState } from 'react';
+import classes from './DeployPopover.module.css';
+import { StudioButton, StudioParagraph, StudioPopover, StudioSpinner } from '@studio/components';
+import { useTranslation } from 'react-i18next';
+
+export type DeployPopoverProps = {
+  appDeployedVersion: string;
+  selectedImageTag: string;
+  disabled: boolean;
+  isPending: boolean;
+  onConfirm: () => void;
+};
+
+export const DeployPopover = ({
+  appDeployedVersion,
+  selectedImageTag,
+  disabled,
+  isPending,
+  onConfirm,
+}: DeployPopoverProps): ReactElement => {
+  const { t } = useTranslation();
+
+  const [isConfirmDeployDialogOpen, setIsConfirmDeployDialogOpen] = useState<boolean>();
+
+  return (
+    <StudioPopover variant='warning' placement='right' open={isConfirmDeployDialogOpen}>
+      <StudioPopover.Trigger
+        disabled={!selectedImageTag || disabled}
+        onClick={() => setIsConfirmDeployDialogOpen((prevState) => !prevState)}
+        size='sm'
+      >
+        {isPending && (
+          <StudioSpinner
+            variant='interaction'
+            size='xsmall'
+            data-testid='spinner-test-id'
+            spinnerTitle=''
```

Comment:
Should we use the title attribute here instead of data-testid for accessibility and clarity? If there’s a reason we prefer not to provide a meaningful spinnerTitle, could you clarify why?

Reply:
I can do that, but again, this was copied code